yksmp/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::ffi::c_void;
+use std::{ptr, slice};
+
+/// Parses LLVM stackmaps version 3 from a given address. Provides a way to query relevant
+/// locations given the return address of a `__llvm_deoptimize` function.
+/// Note that LLVM currently only supports stackmaps for 64 bit architectures. Once they support
+/// others we will need to either make this parser more dynamic or create a new one for each
+/// architecture.
+struct StackMapParser<'a> {
+    data: &'a [u8],
+    offset: usize,
+}
+
+struct Function {
+    addr: u64,
+    record_count: u64,
+}
+
+struct Record {
+    offset: u32,
```

Comment:
Are the `offset`s 32 bits on all platforms?

Reply:
LLVM stackmaps currently only support x64 architectures. So yes.